src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs

```diff
@@ -146,11 +146,7 @@ internal static class BinderHelper
 
         public static void ValidateBindArgument(DynamicMetaObject argument, string paramName)
         {
-            if (argument == null)
-            {
-                throw Error.ArgumentNull(paramName);
-            }
-
+            ArgumentNullException.ThrowIfNull(argument, paramName);
             if (!argument.HasValue)
```

Comment:
Can you remind me why `!!` isn't valid here?  I feel like the strong plurality of calling ArgumentNullException.ThrowIfNull previously was about exception ordering.

Reply:
The parameter name to be used in any resulting exception is getting passed in as the second argument.  Longer term, the right change is probably to annotate all call sites using ValidateBindArgument with !!, and then remove the null checking from here.

Reply:
Ah yes.  I feel like my only legitimate response is "D'oh!" (or, as it's spelled in the scripts: "(annoyed grunt)")